changelog.d/gh-10039.fixed

```diff
@@ -0,0 +1,5 @@
+In generic mode (default, spacegrep engine), matching a pattern that
+ends with an ellipsis now favors the longest match rather than the shortest
+match when multiple matches are possible. For example, for a given target
+program `a a b`, the pattern `a ... b` will match `a b` as before but
+the pattern `a ...` will now match the longer `a a b` rather than `a b`.
```

Comment:
Shouldn't it match _both_ `a`s like Semgrep would do? I know that in Spacegrep `...` only produces the shortest match, but in this case `a ...` can be matched _starting_ at two different locations, shouldn't we match both?

Reply:
> So if I understand correctly, we are getting both matches "a a b" and "a b" and we're keeping the shortest one only because w don't have two matches ending at the same location.

Not sure what you mean by "only because we don't have two matches ending at the same location". We always prefer the shortest match when a true ellipsis is involved (not a leading or trailing ellipsis).